src/app/banking/vouchers/search-bought-vouchers.component.ts

```diff
@@ -0,0 +1,84 @@
+import { BaseSearchPageComponent } from 'app/shared/base-search-page.component';
+import { Component, ChangeDetectionStrategy, Injector, OnInit } from '@angular/core';
+import { UserVouchersDataForSearch, VoucherResult, VoucherRelationEnum, VoucherStatusEnum } from 'app/api/models';
+import { HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { VouchersService } from 'app/api/services';
+import { UserVouchersQueryFilters } from 'app/api/models/user-vouchers-query-filters';
+import { cloneDeep } from 'lodash';
+import { FieldOption } from 'app/shared/field-option';
+
+type UserVouchersSearchParams = UserVouchersQueryFilters & { user: string };
+
+@Component({
+  selector: 'search-bought-vouchers',
+  templateUrl: 'search-bought-vouchers.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SearchBoughtVouchersComponent
+  extends BaseSearchPageComponent<UserVouchersDataForSearch, UserVouchersSearchParams, VoucherResult> implements OnInit {
+
+  user: string;
+  constructor(injector: Injector, private vouchersService: VouchersService) {
+    super(injector);
+  }
+
+  ngOnInit() {
+    super.ngOnInit();
+    const params = this.route.snapshot.params;
+    this.user = this.authHelper.isSelf(params.user) ? this.ApiHelper.SELF : params.user;
+
+    this.addSub(
+      this.vouchersService.getUserVouchersDataForSearch({ user: this.user, relation: VoucherRelationEnum.BOUGHT })
+        .subscribe(data => {
+          this.form.patchValue(data.query, { emitEvent: false });
```

Comment:
I did a change yesterday on master that makes it unnecessary to use the `emitEvent: false` option.
Now the listener for change is added after the full initialization, so there's no risk of searching twice on initialization.